File: Algorithm/python/BAEKJOON/CLASS 2/1920.py
# TRY 1
N = int(input())
A = list(map(int, input().split(" ")))
M = int(input())
M_numbers = list(map(int, input().split(" ")))

A_sorted = sorted(A)

# M_numbers를 원소별로 탐색
for M_number in M_numbers:
    start, end = 0, N - 1 # start와 end는 A List의 index.
    isExist = False

    # 이분탐색 시작
    while start <= end:
        middle = (start + end) // 2 # middle은 A List의 index
        if M_number == A_sorted[middle]:
            isExist = True
            print(1)
            break
        elif M_number > A_sorted[middle]:
            start = middle + 1
        else:
            end = middle - 1

    if not isExist:
        print(0)

File: Algorithm/javascript/BAEKJOON/CLASS 2/1920.js
// TRY 1
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];
rl.on("line", (line) => {
  if (!line) {
    rl.close();
  } else {
    input.push(line);
  }
}).on("close", () => {
  solution(input);
  process.exit();
});

function solution(input) {
  let N = input[0];
  let A = input[1].split(" ");
  let M = input[2];
  let MNumbers = input[3].split(" ");

  let ASorted = A.sort();

  for (let i = 0; i < M; i++) {
    let start = 0;
    let end = N - 1;
    let isExist = false;

    while (start <= end) {
      let middle = parseInt((start + end) / 2);
      if (Number(MNumbers[i]) === Number(ASorted[middle])) {
        isExist = true;
        console.log(1);
        break;
      } else if (Number(MNumbers[i]) > Number(ASorted[middle])) {
        start = middle + 1;
      } else {
        end = middle - 1;
      }
    }
    if (!isExist) {
      console.log(0);
    }
  }
}
